Atlassian.Jira/Remote/IssueFieldService.cs

```diff
@@ -70,20 +70,18 @@ public async Task<IEnumerable<CustomField>> GetCustomFieldsAsync(CustomFieldFetc
             return distinctFields;
         }
 
-        public async Task<IEnumerable<RemoteField>> GetCreateFieldsSchemaForProjectAndIssueTypeAsync(string projectKey, string issueTypeId, CancellationToken token = default(CancellationToken))
+        public async Task<IReadOnlyDictionary<string, IReadOnlyCollection<RemoteField>>> GetCreateFieldsSchemaForProjectAndIssueTypeAsync(string projectKey, string issueTypeId = null, CancellationToken token = default(CancellationToken))
         {
-            var resource = $"rest/api/2/issue/createmeta?expand=projects.issuetypes.fields";
+            var resource = "rest/api/2/issue/createmeta?expand=projects.issuetypes.fields";
 
-            if (!String.IsNullOrEmpty(projectKey))
-            {
-                resource += $"&projectKeys={projectKey}";
-            }
-            else
+            if (string.IsNullOrEmpty(projectKey))
             {
-                throw new InvalidOperationException("Project Key is empty.");
+                throw new InvalidOperationException("Project key is empty.");
             }
 
-            if (!String.IsNullOrEmpty(issueTypeId))
+            resource += $"&projectKeys={projectKey}";

```

Comment:
Does this mean that theoretically we could batch by projects as well?

Reply:
Yes